src/libstd/fs.rs

```diff
@@ -1981,6 +1982,38 @@ mod tests {
     }
 
     #[test]
+    #[cfg(windows)] // only tricky on Windows
+    fn recursive_rmdir_tricky() {
+        use os::windows::fs::OpenOptionsExt;
+        let tmpdir = tmpdir();
+        let dir = tmpdir.join("dir");
+        check!(fs::create_dir(&dir));
+        // filenames that can only be accessed with `/??/`-paths
+        let fullpath = check!(dir.canonicalize());
+        check!(File::create(fullpath.join("morse .. .")));
+        check!(File::create(fullpath.join("con")));
+        // read-only file
+        {
+            let mut opts = fs::OpenOptions::new();
+            opts.write(true);
+            opts.create(true);
+            opts.attributes(0x1); // FILE_ATTRIBUTE_READONLY
```

Comment:
Also if this is the only platform-specific part of this test, couldn't this be a platform-agnostic test by using `set_permissions`?